src/prefect/engine.py

```diff
@@ -1281,14 +1282,26 @@ def link_state_to_result(state: State, result: Any) -> None:
     - The field can be preserved on copy.
     - We cannot set this attribute on Python built-ins.
     """
-    # We cannot track some Python built-ins since they are singletons and could create
-    # confusing relationships, e.g. `None`
-    if type(result) in UNTRACKABLE_TYPES:
-        return
+
+    def link_if_not_untrackable(obj: Any, state: State, ctx: FlowRunContext) -> None:
```

Comment:
This function name is confusing because of the double negative - `link_if_trackable` would be preferable. In addition, this function is being redefined on every call of `link_state_to_result` but I don't believe it's making use of any variables in the closure; it could be defined once as a private function in `engine.py`